src/sentry/middleware/ratelimit.py

```diff
@@ -78,10 +84,17 @@ def _can_be_ratelimited(self, request: Request):
     def process_view(self, request, view_func, view_args, view_kwargs):
         """Check if the endpoint call will violate"""
         if not self._can_be_ratelimited(request):
+            request.will_be_rate_limited = False
             return
 
         key = get_rate_limit_key(view_func, request)
-        request.will_be_rate_limited = (
-            above_rate_limit_check(key)["is_limited"] if key is not None else False
-        )
+        if key is not None:
+            limit, window = get_rate_limit_value(
+                request.method, view_func.view_class, key.split(":", 1)[0]
```

Comment:
The expression `key.split(":", 1)[0]` is a little complicated. Can you toss it in a named variable or function?